src/network.zig

```diff
@@ -546,8 +546,16 @@ pub const ConnectionManager = struct { // MARK: ConnectionManager
 			}
 			if(self.online.load(.acquire) and source.ip == self.externalAddress.ip and source.port == self.externalAddress.port) return;
 		}
-		if(self.newConnectionCallback.load(.monotonic)) |callback| {
-			callback(source);
+		if(self.allowNewConnections.load(.monotonic)) {
+			if(data.len != 0 and data[0] == @intFromEnum(Connection.ChannelId.init)) {
+				const ip = std.fmt.allocPrint(main.stackAllocator.allocator, "{}", .{source}) catch unreachable;
```

Comment:
I am petty so I will point out here that if this is IPv4/v6 then you can easily predict max size of buffer required for that and use `bufPrint`.

Now, so it is not a useless comment, I'd like to point out that there are instances where we use `bufPrint` for eg. paths where length is much harder to predict that makes me ask - what is the guideline here - do we try to converge to one or the other or do we keep that as programmers decision?

Reply:
To contribution guidelines we go, pretty please C: